### PR TITLE
Multivolume storage class fix

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

The previous fix (https://github.com/kubernetes/kubernetes/pull/77525) contained undesired additional changes and fixed the problem at a non-obvious location.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/CC @msau42 @davidz627

### CODE REVIEW

**Comment by pohly** [comment]
/hold

As this problem occurred in a setup that I cannot reproduce myself, this revised fix must be tested manually before merging.

**Comment by msau42** [comment]
/assign @davidz627 
to help test with external pd csi driver

File: test/e2e/storage/external/external.go [hunk 1]
```diff
@@ -242,6 +243,9 @@ func (d *driverDefinition) GetDynamicProvisionStorageClass(config *testsuites.Pe
 
 	sc, ok := items[0].(*storagev1.StorageClass)
 	gomega.Expect(ok).To(gomega.BeTrue(), "storage class from %s", d.StorageClass.FromFile)
+	// Ensure that we can load more than once as required for
+	// GetDynamicProvisionStorageClass by adding a random suffix.
+	sc.Name = names.SimpleNameGenerator.GenerateName(sc.Name)
```
Comment (davidz627) [inline test/e2e/storage/external/external.go]: Generate name will directly append a 5 character random string to the end of the input. Could you add a `-` character between the original sc.Name and the random string for readability?

Comment (pohly) [reply test/e2e/storage/external/external.go]: Fixed and force-pushed.

**Comment by pohly** [comment]
Has someone tested this? If yes, please remove the hold.

If not, then I could try to test by deploying the hostpath driver.

**Comment by davidz627** [comment]
working on testing now

**Comment by davidz627** [comment]
looks like its good
/hold cancel

File: test/e2e/storage/testsuites/testdriver.go [hunk 2]
```diff
@@ -88,6 +88,9 @@ type PreprovisionedPVTestDriver interface {
 type DynamicPVTestDriver interface {
 	TestDriver
 	// GetDynamicProvisionStorageClass returns a StorageClass dynamic provision Persistent Volume.
+	// The StorageClass must be created in the current test's namespace and have
+	// a unique name inside that namespace because GetDynamicProvisionStorageClass might
+	// be called more than once per test.
```
Comment (pohly) [inline test/e2e/storage/testsuites/testdriver.go]: BTW, I got curious why we now have this requirement. It found that it was introduced two months ago when implementing the multi-attach test cases, but then wasn't treated as the API change for the TestDriver interface but rather got handled only in some of its implementations (those that use `GetStorageClass`):
https://github.com/kubernetes/kubernetes/commit/b4c88acec63ebd678d2e5a7a3db1b6da22d39b96

**Pull request merged**